jobbergate-api/jobbergate_api/storage.py

```diff
@@ -81,9 +81,26 @@ def sort_clause(
             status_code=status.HTTP_400_BAD_REQUEST,
             detail=f"Invalid sorting column requested: {sort_field}. Must be one of {sort_field_names}",
         )
-    sort_column: typing.Union[Column, UnaryExpression] = sortable_fields[index]
-    if not sort_ascending:
-        sort_column = sort_column.desc()
+    sort_column: typing.Union[Column, UnaryExpression, Case] = sortable_fields[index]
+    if isinstance(sort_column.type, sqlalchemy.Enum):
+        # SQLAlchemy will not sort enums alphabetically be default, but rather by creation order.
+        # Thus, we have to force alphabetical sorting using a CASE clause.
+        # The logic here is pretty insane. Basically, we have to provide a lookup and then a string
+        # Sort value. If sort order is ascending, each enum value's sort value is itself. Otherwise,
+        # the sort value is it's opposite in the list.
+        # To begin, See: https://stackoverflow.com/a/23618085/642511
+        sorted_values = sorted(sort_column.type.enums)
+        if sort_ascending:
+            sort_tuple = zip(sorted_values, sorted_values)
+        else:
+            sort_tuple = zip(sorted_values, reversed(sorted_values))
+        sort_column = sqlalchemy.case(
+            {key: sort_value for (key, sort_value) in sort_tuple},
+            value=sort_column,
+        )
+    else:
+        if not sort_ascending:
+            sort_column = sort_column.desc()
```

Comment:
as you said, a bit insane 😂 Why not to use the `sorted` built-in function after the data has been fetched from the database? Definitely it would become easier to understand. About performance, well, I might be wrong, but for operations against less than a hundred thousands, it wouldn't be that bad.

Example of what I do using the `sorted` function: https://github.com/omnivector-solutions/armada-admin-api/blob/main/armada_admin_api/routers/organizations/organizations.py#L160

Reply:
We could use that approach, but this is a minimal change that fixes the sorting on enum columns without disrupting the rest of the framework built around sorting that's already in place and functioning as expected.

Additionally, this is a good point of reference if I ever have to sort an enum column again! =]

Reply:
Broke this logic out into its own function with more explanation in commit 616b81b.